Source/engine/random.hpp

```diff
@@ -10,9 +10,135 @@
 #include <algorithm>
 #include <cstdint>
 #include <initializer_list>
+#include <random>
 
 namespace devilution {
 
+class DiabloGenerator {
+private:
+	/** Borland C/C++ psuedo-random number generator needed for vanilla compatibility */
+	std::linear_congruential_engine<uint32_t, 0x015A4E35, 1, 0> lcg;
+
+public:
+	/**
+	 * @brief Set the state of the RandomNumberEngine used by the base game to the specific seed
+	 * @param seed New engine state
+	 */
+	DiabloGenerator(uint32_t seed)
+	{
+		lcg.seed(seed);
+	}
+
+	/**
+	 * @brief Advance the global RandomNumberEngine state by the specified number of rounds
+	 *
+	 * Only used to maintain vanilla compatibility until logic requiring reproducable random number generation is isolated.
+	 * @param count How many values to discard
+	 */
+	void discardRandomValues(unsigned count)
+	{
+		while (count != 0) {
+			static_cast<void>(lcg()); // for avoiding discarded return value warning
+			count--;
+		}
```

Comment:
```suggestion
		lcg.discard(count);
```

Since this class doesn't track or expose the current state there's no real reason to avoid using the built-in discard,